package/kedro_viz/models/flowchart.py

```diff
@@ -577,6 +580,7 @@ def __post_init__(self, data_node: DataNode):
         self.type = data_node.dataset_type
         dataset = cast(AbstractDataset, data_node.kedro_obj)
         dataset_description = dataset._describe()
+        self.code = extract_data_source(self.type, dataset_description)
```

Comment:
can we not follow the approach we have for other datasets where we add dataset specific properties. 

if data_node.is_sql_node():
    self.query = extract_query(data_node)

Reply:
We can, but my concern is that it'll take too long - some refactoring would be needed, as currently we get:
<img width="685" alt="image" src="https://github.com/kedro-org/kedro-viz/assets/35583819/9ca146cd-8af3-4f2f-bd15-1a58b1b78c75">

due to line 609 below
![image](https://github.com/kedro-org/kedro-viz/assets/35583819/799ad95a-4f39-42c8-a0c5-464ad22b30bb)

What do you think of merging as it is? The solution seems to work reliably, and this solution can be implemented when we refactor the other datasets in?